### PR TITLE
MW-342: Improve performance of OrderDto by utilizing search orderables

### PR DESCRIPTION
**Pull request by jkondrat**

### CODE REVIEW

File: src/main/java/org/openlmis/fulfillment/service/ExporterBuilder.java [hunk 1]
```diff
@@ -78,10 +81,14 @@ public void export(Order order, Order.Exporter exporter) {
...
-  public void export(OrderLineItem item, OrderLineItem.Exporter exporter) {
+  public void export(OrderLineItem item, OrderLineItem.Exporter exporter,
+                     List<OrderableDto> orderables) {
     exporter.setId(item.getId());
     exporter.setApprovedQuantity(item.getApprovedQuantity());
-    exporter.setOrderable(getIfPresent(products, item.getOrderableId()));
+    Optional<OrderableDto> orderableDto = orderables.stream().filter(
+        orderable -> orderable.getId().equals(item.getOrderableId())).findAny();
+    exporter.setOrderable(orderableDto.orElse(
+            products.findOne(item.getOrderableId())));
```
Comment (palbecki) [inline src/main/java/org/openlmis/fulfillment/service/ExporterBuilder.java]: why don't you use getIfPresent method anymore?

File: src/main/java/org/openlmis/fulfillment/service/ExporterBuilder.java [hunk 2]
```diff
@@ -130,6 +137,17 @@ public void export(FtpTransferProperties properties,
     return dtos;
   }
 
+  /**
+   * Fetch orderables for each line item of given order.
+   * @param order related order
+   * @return a list of orderable dtos
+   */
+  public List<OrderableDto> getLineItemOrderables(Order order) {
```
Comment (palbecki) [inline src/main/java/org/openlmis/fulfillment/service/ExporterBuilder.java]: Can you add test for that method?

File: src/main/java/org/openlmis/fulfillment/service/ExporterBuilder.java [hunk 3]
```diff
@@ -78,10 +81,14 @@ public void export(Order order, Order.Exporter exporter) {
...
    * {@link OrderLineItem.Exporter} interface.
    */
-  public void export(OrderLineItem item, OrderLineItem.Exporter exporter) {
+  public void export(OrderLineItem item, OrderLineItem.Exporter exporter,
+                     List<OrderableDto> orderables) {
     exporter.setId(item.getId());
     exporter.setApprovedQuantity(item.getApprovedQuantity());
-    exporter.setOrderable(getIfPresent(products, item.getOrderableId()));
+    Optional<OrderableDto> orderableDto = orderables.stream().filter(
+        orderable -> orderable.getId().equals(item.getOrderableId())).findAny();
```
Comment (palbecki) [inline src/main/java/org/openlmis/fulfillment/service/ExporterBuilder.java]: Can you add test that filter?

File: src/main/java/org/openlmis/fulfillment/service/ExporterBuilder.java (hunk 2 above)
Comment (jkondrat) [inline src/main/java/org/openlmis/fulfillment/service/ExporterBuilder.java]: Done

File: src/main/java/org/openlmis/fulfillment/service/ExporterBuilder.java (hunk 3 above)
Comment (jkondrat) [inline src/main/java/org/openlmis/fulfillment/service/ExporterBuilder.java]: Done

**Pull request merged**